pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1184,6 +1239,40 @@ public void redeliverUnacknowledgedMessages(Set<MessageId> messageIds) {
             MessageIdData.Builder builder = MessageIdData.newBuilder();
             batches.forEach(ids -> {
                 List<MessageIdData> messageIdDatas = ids.stream().map(messageId -> {
+                    List<MessageImpl<T>> deadLetterMessages = null;
```

Comment:
Can this new portion of code be factored out in a separate method that is called from here?

Reply:
ok